charts/bindplane/templates/bindplane-nats.yaml

```diff
@@ -166,7 +168,12 @@ spec:
             - name: BINDPLANE_NATS_SERVER_CLUSTER_PORT
               value: "6222"
             - name: BINDPLANE_NATS_SERVER_CLUSTER_ROUTES
+              {{ if eq .Values.nats.deploymentType "StatefulSet" }}
               value: nats://{{ include "bindplane.fullname" . }}-nats-0.{{ include "bindplane.fullname" . }}-nats-cluster-headless.{{ .Release.Namespace }}.svc.cluster.local:6222,nats://{{ include "bindplane.fullname" . }}-nats-1.{{ include "bindplane.fullname" . }}-nats-cluster-headless.{{ .Release.Namespace }}.svc.cluster.local:6222,nats://{{ include "bindplane.fullname" . }}-nats-2.{{ include "bindplane.fullname" . }}-nats-cluster-headless.{{ .Release.Namespace }}.svc.cluster.local:6222
+              {{ end }}
+              {{ if eq .Values.nats.deploymentType "Deployment" }}
+              value: nats://{{ include "bindplane.fullname" . }}-nats-cluster-headless.{{ .Release.Namespace }}.svc.cluster.local:6222
+              {{ end }}
```

Comment:
StatefulSet pod hostnames are predictable when the replica count is a fixed number, which is why we can use them.

When operating as a Deployment, targetnig the headless service works fine.